apollo-router/src/configuration/mod.rs

```diff
@@ -143,6 +143,11 @@ impl Configuration {
 
         plugins
     }
+
+    //. checks that we can reload configuration from the current one to the new one
+    pub fn is_compatible(&self, new: &Configuration) -> bool {
```

Comment:
maybe this should return a `Result`, because we might have other errors in the future? (and we could set the error message here)